lib/logstash_sidekiq_logger.rb

```diff
@@ -0,0 +1,37 @@
+# Sets up a new sidekiq file logger for logstash consumption.
+#
+# To do this it combines data from ActiveJob notifications and from Sidekiq
+# logs.
+#
+# ActiveJob notifications provides data on what job ran with what parameters,
+# this is obtained by setting up a subscription to the ActiveJob instrumentaton
+# that Rails provides.
+#
+# Sidekiq provides no instrumentation so the ActiveJob data is combined on a
+# custom log formatter.
+#
+# The final output looks like this:
+#
+# rubocop:disable LineLength
+# {"job_name":"prison_mailer_request_received","arguments":["gid://prison-visits/Visit/0125d92d-3e27-4835-bc35-7fd5fb75033e"],"queue_name":"mailers","status":"completed","active_job_duration":640.827,"total_duration":691.0,"@timestamp":"2016-06-17T13:22:58.096Z","@version":"1"}
+# {"job_name":"prison_mailer_request_received","arguments":["gid://prison-visits/Visit/0125d92d-3e27-4835-bc35-7fd5fb75033e"],"queue_name":"mailers","status":"to_be_retried","active_job_duration":1.194,"total_duration":4.0,"retry_number":1,"@timestamp":"2016-06-17T13:00:26.298Z","@version":"1"}
+# rubocop:enable LineLength
```

Comment:
What're these for?

If it's for readers of this documentation to see a realistic sample of the fields in the log lines, I might be tempted to break it over multiple lines so it's easier to see which bits are keys and which are values (and that way, you can remove the `rubocop:disable`). That said, I worry that this will become out of sync with the actual code, particularly given that the code is in an entirely other file.

Maybe instead say something like “Final output contains the fields specified in LogstashSidekiqLogger.Formatter#base_log_data, compressed onto one line”?